packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1046,16 +1046,14 @@ function updateMemo<T>(
   return nextValue;
 }
 
-// in a test-like environment, we want to warn if dispatchAction()
-// is called outside of a batchedUpdates/TestUtils.act(...) call.
-let shouldWarnForUnbatchedSetState = false;
+let shouldWarnForUnactedSetState = false;
 
 if (__DEV__) {
   // jest isn't a 'global', it's just exposed to tests via a wrapped function
   // further, this isn't a test file, so flow doesn't recognize the symbol. So...
   // $FlowExpectedError - because requirements don't give a damn about your type sigs.
   if ('undefined' !== typeof jest) {
-    shouldWarnForUnbatchedSetState = true;
+    shouldWarnForUnactedSetState = true;
```

Comment:
This doesn't need to be a module-level variable. Make this a local check right before firing the warning.